src/Bundle/Resources/skeleton/Factory.tpl.php

```diff
@@ -2,49 +2,54 @@
 
 namespace <?= $namespace; ?>;
 
-use <?= $entity->getName() ?>;
-<?php if ($repository): ?>use <?= $repository->getName() ?>;
+use <?= $object->getName() ?>;
+<?php if ($persisted && $repository): ?>use <?= $repository->getName() ?>;
 use Zenstruck\Foundry\RepositoryProxy;
 <?php endif ?>
 use Zenstruck\Foundry\ModelFactory;
 use Zenstruck\Foundry\Proxy;
 
 /**
- * @extends ModelFactory<<?= $entity->getShortName() ?>>
+ * @extends ModelFactory<<?= $object->getShortName() ?>>
  *
- * @method <?= $entity->getShortName() ?>|Proxy create(array|callable $attributes = [])
- * @method static <?= $entity->getShortName() ?>|Proxy createOne(array $attributes = [])
- * @method static <?= $entity->getShortName() ?>|Proxy find(object|array|mixed $criteria)
- * @method static <?= $entity->getShortName() ?>|Proxy findOrCreate(array $attributes)
- * @method static <?= $entity->getShortName() ?>|Proxy first(string $sortedField = 'id')
- * @method static <?= $entity->getShortName() ?>|Proxy last(string $sortedField = 'id')
- * @method static <?= $entity->getShortName() ?>|Proxy random(array $attributes = [])
- * @method static <?= $entity->getShortName() ?>|Proxy randomOrCreate(array $attributes = [])
- * @method static <?= $entity->getShortName() ?>[]|Proxy[] all()
- * @method static <?= $entity->getShortName() ?>[]|Proxy[] createMany(int $number, array|callable $attributes = [])
- * @method static <?= $entity->getShortName() ?>[]|Proxy[] createSequence(array|callable $sequence)
- * @method static <?= $entity->getShortName() ?>[]|Proxy[] findBy(array $attributes)
- * @method static <?= $entity->getShortName() ?>[]|Proxy[] randomRange(int $min, int $max, array $attributes = [])
- * @method static <?= $entity->getShortName() ?>[]|Proxy[] randomSet(int $number, array $attributes = [])
+ * @method <?= $object->getShortName() ?>|Proxy create(array|callable $attributes = [])
+ * @method static <?= $object->getShortName() ?>|Proxy createOne(array $attributes = [])
+<?php if ($persisted): ?> * @method static <?= $object->getShortName() ?>|Proxy find(object|array|mixed $criteria)
```

Comment:
I was wondering if we should not split into two template files, but I think we should not: otherwise we would need to add new factories' methods in both places.

Reply:
Yeah, if it gets out of hand later we can split.